vng/vector/io.go

```diff
@@ -76,7 +76,7 @@ func NewWriter(typ zed.Type, spiller *Spiller) Writer {
 		if !zed.IsPrimitiveType(typ) {
 			panic(fmt.Sprintf("unsupported type in VNG file: %T", typ))
 		}
-		return NewPrimitiveWriter(typ, spiller, true)
```

Comment:
Previously nulls in primitive values would be written as zng nulls in the primitive column rather than producing a nulls column. This was hidden by the vng reader using a zcode.Builder to reconstruct them.

Reply:
This was by design at the time as the Nulls column was used only for complex types.